ptypes/regen.sh

```diff
@@ -54,18 +76,23 @@ done
 # Pass 2: copy files, making necessary adjustments.
 for up in "${!filename_map[@]}"; do
   f=${filename_map[$up]}
-  shortname=$(basename $f | sed 's,\.proto$,,')
+  shortname=$(basename $f | $(which_sed) 's,\.proto$,,')
   cat $tmpdir/$UPSTREAM_SUBDIR/$up |
     # Adjust proto package.
     # TODO(dsymonds): Upstream the go_package option instead.
-    sed '/^package /a option go_package = "'${shortname}'";' |
+    $(which_sed) '/^package /a option go_package = "'github.com\/golang\/protobuf\/ptypes\/${shortname}'";' |
     # Unfortunately "package struct" doesn't work.
-    sed '/option go_package/s,"struct","structpb",' |
+    $(which_sed) '/option go_package/s,struct",struct;structpb",' |
     cat > $PKG/$f
 done
 
 # Run protoc once per package.
-for dir in $(find $PKG -name '*.proto' | xargs dirname | sort | uniq); do
+#
```

Comment:
`xargs -n1 dirname` will run dirname for each file in the output of find (-n is part of the posix xargs standard so should be cross platform).  The implementation below will run multiple times per directory if there are multiple .proto files in that directory